internal/abci/abci.go

```diff
@@ -959,6 +1001,38 @@ func (a *AbciApp) Commit(ctx context.Context, _ *abciTypes.RequestCommit) (*abci
 	return &abciTypes.ResponseCommit{}, nil // RetainHeight stays 0 to not prune any blocks
 }
 
+// lastCommitInfo is a struct to store the last commit info such as
+// height and apphash at the end of the FinalizeBlock method.
+type lastCommitInfo struct {
+	AppHash types.HexBytes `json:"app_hash"`
+	Height  int64          `json:"height"`
+}
+
+// saveAs saves the last commit info to root_dir/abci/last_commit_info.json.
+func (lc *lastCommitInfo) saveAs(filename string) error {
```

Comment:
Just a nit, but these method docs are misleading.  They are not aware of the path the docs mention at all.  They save/load blindly from the *absolute* path provided in `filename`